src/python/pants/engine/isolated_process.py

```diff
@@ -78,15 +78,12 @@ class MultiPlatformProcess:
     platform_constraints: Tuple[str, ...]
     processes: Tuple[Process, ...]
 
-    def __init__(
-        self, request_dict: Dict[Tuple[PlatformConstraint, PlatformConstraint], Process],
-    ) -> None:
+    def __init__(self, request_dict: Dict[PlatformConstraint, Process],) -> None:
         if len(request_dict) == 0:
             raise ValueError("At least one platform constrained Process must be passed.")
         validated_constraints = tuple(
             constraint.value
-            for pair in request_dict.keys()
-            for constraint in pair
+            for constraint in request_dict.keys()
```

Comment:
Might be useful to call this `execution_constraint` now that it is just one type of constraint.